02-oop-inheritance/js/script.js

```diff
@@ -0,0 +1,82 @@
+var ObserverList = function () {
+    this.observerList = [];
+};
+
+ObserverList.prototype = {
+
+    constructor: ObserverList,
+
+    subscribeObserver: function (observer) {
+        this.observerList.push(observer);
+    },
+
+    unsubscribeObserver: function (observer) {
+        this.observerList.splice(this.observerList.indexOf(observer), 1);
+    }
+};
+
+var Movie = function (title,year,observer) {
+    this.title = title;
+    this.year = year;
+    ObserverList.call(observer);
+};
+
+Movie.prototype = {
+
+    constructor: Movie,
+
+    get: function (attribute) {
+        return this[attribute];
+    },
+
+    set: function (attribute, value) {
+        this[attribute] =  value;
+    },
+
+    startplaying: function () {
+        for( i = 0; i < observerList.length ; i++) {
```

Comment:
``observerList`` is not defined anywhere, I think you wanted to point to ``this.observerList``.